example/main.go

```diff
@@ -220,16 +216,17 @@ func mutatePodsFn(pod *corev1.Pod) error {
 // podValidator validates Pods
 type podValidator struct {
 	client client.Client
+	codec  serializer.CodecFactory
 }
 
 // Implement admission.Handler so the controller can handle admission request.
 var _ admission.Handler = &podValidator{}
 
 // podValidator admits a pod iff a specific annotation exists.
-func (a *podValidator) Handle(_ context.Context, req admission.Request) admission.Response {
+func (v *podValidator) Handle(_ context.Context, req admission.Request) admission.Response {
 	pod := &corev1.Pod{}
 
-	err := admission.DecodeRequest(req, pod, nil)
+	err := admission.DecodeRequest(req, pod, v.codec.UniversalDeserializer())
```

Comment:
I don't like requiring the user to manually know about this.  Can we have like a `RequestDecoder` interface that the user can get a handle to?  Like we do with clients...